File: ios/batm/Entities/Transaction/TransactionType.swift
import UIKit

enum TransactionType: Int {
    case unknown = 0
    case deposit = 1
    case withdraw = 2
    case sendTransfer = 3
    case receiveTransfer = 4
    case buy = 5
    case sell = 6
    case move = 7
    case sendSwap = 8
    case receiveSwap = 9
    case reserve = 10
    case recall = 11
    case `self` = 12
    case createStake = 13
    case cancelStake = 14
    case withdrawStake = 15
    
    var verboseValue: String {
        switch self {
        case .unknown: return localize(L.CoinDetails.unknown)
        case .deposit: return localize(L.CoinDetails.deposit)
        case .withdraw: return localize(L.CoinDetails.withdraw)
        case .sendTransfer: return localize(L.CoinDetails.sendGift)
        case .receiveTransfer: return localize(L.CoinDetails.receiveGift)
        case .buy: return localize(L.CoinDetails.buy)
        case .sell: return localize(L.CoinDetails.sell)
        case .move: return localize(L.CoinDetails.move)
        case .sendSwap: return localize(L.CoinDetails.sendSwap)
        case .receiveSwap: return localize(L.CoinDetails.receiveSwap)
        case .reserve: return localize(L.CoinDetails.reserve)
        case .recall: return localize(L.CoinDetails.recall)
        case .self: return localize(L.CoinDetails.se1f)
        case .createStake: return localize(L.CoinDetails.createStake)
        case .cancelStake: return localize(L.CoinDetails.cancelStake)
        case .withdrawStake: return localize(L.CoinDetails.withdrawStake)
        }
    }
    
    var associatedColor: TransactionTypeColorPalette.Type {
        switch self {
        case .unknown: return UIColor.TransactionTypeColor.Gray.self
        case .deposit: return UIColor.TransactionTypeColor.Green.self
        case .withdraw: return UIColor.TransactionTypeColor.Yellow.self
        case .sendTransfer: return UIColor.TransactionTypeColor.Yellow.self
        case .receiveTransfer: return UIColor.TransactionTypeColor.Green.self
        case .buy: return UIColor.TransactionTypeColor.Green.self
        case .sell: return UIColor.TransactionTypeColor.Yellow.self
        case .move: return UIColor.TransactionTypeColor.Blue.self
        case .sendSwap: return UIColor.TransactionTypeColor.Yellow.self
        case .receiveSwap: return UIColor.TransactionTypeColor.Green.self
        case .reserve: return UIColor.TransactionTypeColor.Yellow.self
        case .recall: return UIColor.TransactionTypeColor.Green.self
        case .self: return UIColor.TransactionTypeColor.Blue.self
        case .createStake: return UIColor.TransactionTypeColor.Yellow.self
        case .cancelStake: return UIColor.TransactionTypeColor.Blue.self
        case .withdrawStake: return UIColor.TransactionTypeColor.Green.self
        }
    }
}

File: app/src/main/java/com/belcobtm/domain/transaction/type/TransactionType.kt
package com.belcobtm.domain.transaction.type

import com.belcobtm.R

enum class TransactionType {
    DEPOSIT,
    WITHDRAW,
    SEND_TRANSFER,
    RECEIVE_TRANSFER,
    ATM_BUY,
    ATM_SELL,
    MOVE,
    SEND_SWAP,
    RECEIVE_SWAP,
    RESERVE,
    RECALL,
    SELF,
    CREATE_STAKE,
    CANCEL_STAKE,
    WITHDRAW_STAKE,
    BUY,
    SELL,
    UNKNOWN; // exists only on client to handle null

    fun getResText(): Int = when (this) {
        DEPOSIT -> R.string.transaction_type_deposit
        WITHDRAW -> R.string.transaction_type_withdraw
        SEND_TRANSFER -> R.string.transaction_type_send_transfer
        RECEIVE_TRANSFER -> R.string.transaction_type_receive_transfer
        ATM_BUY -> R.string.transaction_type_atm_buy
        ATM_SELL -> R.string.transaction_type_atm_sell
        MOVE -> R.string.transaction_type_move
        SEND_SWAP -> R.string.transaction_type_send_swap
        RECEIVE_SWAP -> R.string.transaction_type_receive_swap
        RESERVE -> R.string.transaction_type_reserve
        RECALL -> R.string.transaction_type_recall
        SELF -> R.string.transaction_type_self
        CREATE_STAKE -> R.string.transaction_type_create_stake
        CANCEL_STAKE -> R.string.transaction_type_cancel_stake
        WITHDRAW_STAKE -> R.string.transaction_type_withdraw_stake
        BUY -> R.string.transaction_type_buy
        SELL -> R.string.transaction_type_sell
        UNKNOWN -> R.string.transaction_type_unknown
    }

}
